src/widget.js

```diff
@@ -135,15 +135,16 @@ Widget.prototype = {
       this.rsWidget.classList.add(`rs-state-${state || this.state}`);
       if (this.closed && state !== 'close') {
         this.rsWidget.classList.add('rs-state-close');
+      } else if (!this.closed) {
+        this.rsWidget.classList.remove('rs-state-close');
```

Comment:
I don't see how this makes sense. The current class is already being removed in line 134, so why do we have to remove it again? This being unnecessary would also explain why the `else if` just looks strange when reading it without context.

Reply:
Yes, the `rs-state-close` exists in addition to the other state (e.g. `rs-state-connected` or `rs-state-offline`. I did not code the state system, so I don't know how it was intended. But how else would you indicate that the widget is offline AND closed?

Reply:
Ok, I see. But that still doesn't explain the code. Also, it's something we should probably change right now, so that it's both cleaner and easier to understand and maintain.

Reply:
Yes, I think it would be better to not handle the closing of the widget using a state class, but a different class (e.g. `rs-closed`) instead. That way we can remove the logic from the `state()` function and just add or remove the class from the `open()` and `close()` functions.